cli/internal/cmd/trackingplan/apply/apply.go

```diff
@@ -155,6 +155,20 @@ func createResourceGraph(catalog *localcatalog.DataCatalog) *resources.Graph {
 				Config:      prop.Config,
 			}
 
+			if strings.HasPrefix(prop.Type, "#/custom-types/") {
+				customTypeURN := getURNFromRef(prop.Type)
+
+				if customTypeURN != "" {
+					log.Debug("property references custom type", "property", prop.LocalID, "customTypeRef", prop.Type, "customTypeURN", customTypeURN)
+					args.Type = resources.PropertyRef{
```

Comment:
Having Type as any is troubling, especially if the actually type doesn't depend on the actual property configurations, it depends at which point we parse it. We start by expecting either a string type (with a simple property type) or a custom-type reference, but later  (during create and update) we treat the same args as having either the simple property type or the custom-type's name.

Post this version, (or as part of making custom types support GA) we need to revisit how we handle args, state and resource data.

Reply:
Yeah I agree. But wouldn't this be true of any reference. When we create the entity to hold a reference it needs to be of any type right ?
To respect both the resolved type and PropertyRef type. I wasn't able to find a better way to do conversion that this and would like your opinion on how can we improve it.